catalyst_voices/packages/internal/catalyst_voices_blocs/lib/src/workspace/workspace_bloc.dart

```diff
@@ -0,0 +1,115 @@
+import 'package:catalyst_voices_blocs/src/workspace/workspace_event.dart';
+import 'package:catalyst_voices_blocs/src/workspace/workspace_state.dart';
+import 'package:catalyst_voices_models/catalyst_voices_models.dart';
+import 'package:catalyst_voices_services/catalyst_voices_services.dart';
+import 'package:catalyst_voices_view_models/catalyst_voices_view_models.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+final class WorkspaceBloc extends Bloc<WorkspaceEvent, WorkspaceState> {
+  final CampaignService _campaignService;
+
+  final _answers = <SectionStepId, MarkdownString>{};
+  final _guidances = <SectionStepId, List<Guidance>>{};
+
+  SectionStepId? _activeStepId;
+
+  WorkspaceBloc(
+    this._campaignService,
+  ) : super(const WorkspaceState()) {
+    on<LoadCurrentProposalEvent>(_loadCurrentProposal);
+    on<UpdateStepAnswerEvent>(_updateStepAnswer);
+    on<ActiveStepChangedEvent>(_handleActiveStepEvent);
+  }
+
+  Future<void> _loadCurrentProposal(
+    LoadCurrentProposalEvent event,
+    Emitter<WorkspaceState> emit,
+  ) async {
+    _answers.clear();
+    _guidances.clear();
+
+    final activeCampaign = await _campaignService.getActiveCampaign();
+    if (activeCampaign == null) {
+      emit(
+        state.copyWith(
+          sections: [],
+          guidance: const WorkspaceGuidance(isNoneSelected: true),
+        ),
+      );
+      return;
+    }
+
+    final template = activeCampaign.proposalTemplate;
+
+    final sections = template.sections.map(
+      (section) {
+        return WorkspaceSection(
```

Comment:
Could this be a separate method? The mapping. The `_loadCurrentProposal` method is quite long because of it

Reply:
yep, makes sense